### PR TITLE
Remove dfusion, df2mc, mapexport, dwarfexport

### PR DESCRIPTION
**Pull request by PeridexisErrant**
See #735 - these plugins have not been built since 40.01 (or earlier), and nobody missed them.  DFusion *has* been built, but @warmist notes that almost all functions have been implemented by more recent tools (and multirace embarks is on the issue tracker).

This is mostly cleaning out unused code, with a side of making the project structure easier to understand for users and developers.

Only dfusion is noted as removed in NEWS.rst, as the others have already been removed from the release packages.

### CODE REVIEW

**Comment by BenLubar** [comment]
Doesn't dfusion control the `lua` command? Or is that dead code?

**Comment by lethosor** [comment]
No, that's `scripts/lua.lua`. `scripts/dfusion.lua` is the current dfusion front-end.

**Comment by JapaMala** [comment]
mapexport is used for fortress overseer, but I don't know if anybody still
uses it.

**Comment by lethosor** [comment]
Right now, I'm leaning towards removing df2mc and dfusion (waiting to hear back from @warmist on the latter). dwarfexport and mapexport might still be useful, at least until everything they implement is also implemented by RemoteFortressReader (which @JapaMala would know better than I would), so I'm leaning towards repairing them if that's possible.

**Comment by warmist** [comment]
Just to have same info here as in irc. Dfusion is mostly outdated. Most important thing that i could think of is the site creation that should probably be moved to mod-tools.

File: NEWS.rst [hunk 1]
```diff
@@ -36,6 +36,10 @@ Fixes
 -----
 - confirm haul-delete: Fixed issue preventing deletion of stop conditions or using "x" in route names
 
+Removed
+-------
+- DFusion: legacy script system, obsolete or replaced by better alternatives
```
Comment (lethosor) [inline NEWS.rst]: Wrong section

**Comment by lethosor** [comment]
Okay, so it turns out that the NEWS.rst entries here and in #754 wound up in the wrong section because the commits were made before 0.40.24-r5. Sorry about that.
Maybe rebasing old PRs in the future would help, but at the very least we should verify that changelog entries end up in the right place when merging.

**Comment by JapaMala** [comment]
I don't know anything about dwarfexport, but mapexport is used by Fortress Overseer, which can't use anything else. But since the overseer project is abandoned, as far as I can see, we might be able to get rid of it unless somebody feels like updating it.

**Comment by PeridexisErrant** [comment]
(Pull already merged, but here's some rationale again)

[fortress overseer](http://dffd.bay12games.com/file.php?id=3882) hasn't worked since 34.11 - it can't handle eg multitile trees - and was last updated in 2012.  Plus the plugin hasn't compiled for ages.

`dwarfexport` was only used by [Runesmith](http://dffd.bay12games.com/file.php?id=2538), a Therapist-precursor with hints of `gm-editor`.  Outdated since 31.25!

`df2mc` only exported valid levels for Minecraft Alpha, and again hadn't built in ages.

`dfusion` is the most questionable case, but is clearly legacy code.  Having been mostly replaced by better things, it's time to take it out.

**Pull request merged**